core/trino-main/src/main/java/io/trino/execution/RemoteTaskFactory.java

```diff
@@ -35,5 +37,6 @@ RemoteTask createRemoteTask(
             OptionalInt totalPartitions,
             OutputBuffers outputBuffers,
             PartitionedSplitCountTracker partitionedSplitCountTracker,
+            Map<PlanNodeId, DynamicFilter> dynamicFilters,
```

Comment:
i'm extracting `dynamicFilters` in SqlStageExecution and passing it down into RemoteTaskFactory to avoid repeating that work on creation of every task.